widget/map.go

```diff
@@ -29,11 +29,22 @@ type Map struct {
 	zoom, x, y int
 
 	cl *http.Client
+
+	TileSource          string // url to download xyz tiles (example: https://tile.openstreetmap.org/%d/%d/%d.png)
+	EnableOsmDisclaimer bool   // enable osm copyright disclaimer
+	EnableZoomButtons   bool   // enable zoom buttons
+	EnableMoveButtons   bool   // enable move map buttons
```

Comment:
I think making those field directly accessible and public might be a bit problematic as they could be modified during the life of the widget with unexpected result for the user.

I would think introducing a Config{} or Options{} structure and a new NewMapWithConfig would be a better design pattern. In which case I would make also the http.Client{} part of that structure too.

Reply:
This is a cool idea. Another possibility would be to just use setters.

If you go with public fields (even if it's an Options struct) then remember zero values are the normal defaults - so if it should be visible by default then maybe the field is `HideZoomControls`

Reply:
Sounds good. I like the idea that the configuration and the widget are separated. Now that I think of, maybe the options-pattern could be useful for this use case. 

Something like this:

```go
type MapOption func(*Map)

func WithOsmTiles() MapOption {
	return func(m *Map) {
		m.tileSource = "https://tile.openstreetmap.org/%d/%d/%d.png"
		m.enableOsmDisclaimer = true
	}
}

func NewMapWithOptions(opts ...MapOption) *Map {
	// defaults
	m := &Map{
		cl:                  &http.Client{},
		tileSource:          "https://tile.openstreetmap.org/%d/%d/%d.png",
		enableOsmDisclaimer: true,
		EnableZoomButtons:   true,
		EnableMoveButtons:   true,
	}
	m.ExtendBaseWidget(m)
	for _,opt := range opts {
		opt(m)
	}
	return m
}

m := xwidget.NewMapWithOptions(
	xwidget.WithOsmTiles(),
)
```